src/User/ProfilePage.jsx

```diff
@@ -29,17 +38,47 @@ const ProfilePage = ({ error, isLoading, user }) => {
     return <p>An error has occurred...</p>;
   }
 
-  const { id, username, firstName, lastName, email, occupation } = user || {};
+  const { id, username, firstName, lastName, email, occupation, managerId } =
+    user || {};
+
+  const displayManager = () => {
+    if (String(managerName) === "None Assigned") {
+      return <p className="manager">{`Manager: ${managerName}`}</p>;
+    } else {
+      return (
+        <p className="manager">
+          <>
+            {"Manager: "}
+            <StyledLink
+              className="managerLink"
+              onClick={() => {
+                setIsCurrentUser(false);
+                setManagerName("");
+              }}
+              to={`/users/${managerId}`}
+            >
+              {managerName}
+            </StyledLink>
+          </>
+        </p>
+      );
+    }
+  };
 
   return (
     <StyledCard>
-      <StyledLink className="editLink" to={`/users/edit/${id}`}>Edit</StyledLink>        
+      {isCurrentUser && (
+        <StyledEditLink className="editLink" to={`/users/${id}/edit`}>
+          Edit
+        </StyledEditLink>
+      )}
       <StyledCardHeading className="name">
         {`Hello, ${firstName} ${lastName}!`}
       </StyledCardHeading>
       <p className="username">{`Username: ${username}`}</p>
       <p className="email">{`Email: ${email}`}</p>
       <p className="occupation">{`Occupation: ${occupation}`}</p>
+      {managerName && displayManager()}
```

Comment:
Should look into useMemo hook here. This function is going to be computed every time the component is rendered even though the value wont change. If you memoize the function, the value will be cached and only recomputed when a dependancy is updated (ie managerName)

Reply:
https://reactjs.org/docs/hooks-reference.html#usememo